### PR TITLE
Fixes for Nanbield

### PR DESCRIPTION
**Pull request by shr-project**

### CODE REVIEW

**Comment by shr-project** [comment]
lvgl changes aren't complete as luneos-recovery-ui fails to build
```
/OE/build/luneos-nanbield/webos-ports/tmp-glibc/work/aarch64-webos-linux/luneos-recovery-ui/1.0-r0/git/recovery/main.c:40:25: error: 'fbdev_flush'
 undeclared (first use in this function)
/OE/build/luneos-nanbield/webos-ports/tmp-glibc/work/aarch64-webos-linux/luneos-recovery-ui/1.0-r0/git/recovery/main.c:47:29: error: 'evdev_read' 
undeclared (first use in this function)
```
might be my mistake when rebasing config patch or just API changes in new version.

Another part is libgpiod downgrade in meta-webos-ports needed for eg25-manager
https://github.com/webOS-ports/meta-webos-ports/commit/6a7f2cc6e387c567372dd471689dfdf7ffdfc0c4

**Comment by shr-project** [comment]
I've pushed u-boot and trusted-firmware-a changes to langdale, mickledore, master branches, only the last 3 commits remain.

File: recipes-luneui/recovery/luneos-recovery-ui/0001-recovery-update-for-lvgl-v8.3.0-fbdev_get_sizes-lv_t.patch [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
++    uint32_t dpi;
++    fbdev_get_sizes(&disp_drv.hor_res, &disp_drv.ver_res, &dpi);
+     disp_drv.flush_cb = fbdev_flush;
+     lv_disp_drv_register(&disp_drv);
+ 
+@@ -54,7 +55,6 @@ int main(void)
+     while(1) {
+         lv_task_handler();
+         usleep(5000);    // wait 5ms
+-        lv_tick_inc(5);  // inform lvgl that 5ms have passed
```
Comment (Tofee) [inline recipes-luneui/recovery/luneos-recovery-ui/0001-recovery-update-for-lvgl-v8.3.0-fbdev_get_sizes-lv_t.patch:38]: This looks suspicious, the tick API didn't change in newest LVGL. Are you sure?
see https://docs.lvgl.io/8.3/porting/tick.html?highlight=lv_tick_inc

**Review by Tofee — APPROVED** [review]
This looks good to me. Ideally I would test it on device to confirm the recovery UI still works, but we can also do that later.

**Pull request merged**